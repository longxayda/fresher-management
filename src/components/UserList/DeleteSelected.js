import React, { useEffect, useState } from "react";
// react-bootstrap components
import { Modal, Button } from "react-bootstrap";
import NotificationAlert from "react-notification-alert";
import { useDispatch, useSelector } from "react-redux";
import { USER_LIST_SELECTOR } from "redux/selectors/UserListSelectors";
import { deleteMultiUsersData } from "redux/UserListSlices/UserListSlice";

function DeleteSelected({ handleCurrentPage, showModal = false }) {
  const dispatch = useDispatch();
  const selectArr = useSelector(USER_LIST_SELECTOR).selectedUser;
  const [showDeleteUser, setshowDeleteUser] = useState(showModal);
  const handleCloseDeleteUser = () => setshowDeleteUser(false);
  const handleShowDeleteUser = () => setshowDeleteUser(true);
  let isSelect = selectArr.length;
  const handleSaveofDelete = () => {
    dispatch(deleteMultiUsersData(selectArr));
    // handleCurrentPage();
    handleCloseDeleteUser();
    notify("tr");
  };
  // Notification
  const notificationAlertRef = React.useRef(null);
  const notify = (place) => {
    var options = {};
    options = {
      place: place,
      message: (
        <div>
          <div>User has been deleted</div>
        </div>
      ),
      type: "success",
      icon: "nc-icon nc-bell-55",
      autoDismiss: 7,
    };
    notificationAlertRef.current.notificationAlert(options);
  };
  return (
    <>
      <Button
        id="deleteSelectedBtn"
        className="btn-fill"
        variant="danger"
        style={{ marginLeft: 10, width: 130 }}
        onClick={handleShowDeleteUser}
        disabled={!isSelect}
      >
        <i className="fas fa-trash" style={{ marginRight: "5px" }}></i>Delete{" "}
        {isSelect}
      </Button>
      <NotificationAlert ref={notificationAlertRef} />
      <Modal
        show={showDeleteUser}
        onHide={handleCloseDeleteUser}
        centered
        scrollable
        style={{ maxHeight: 700 }}
      >
        <Modal.Header closeButton>
          <Modal.Title>
            <div className="fw-semibold fs-3 text-danger modal-title h4" style={{textTransform:'uppercase'}}>Delete users selected ?</div>
          </Modal.Title>
        </Modal.Header>

        <Modal.Body>
          {selectArr.map((user, index) => {
            return (
              <div id="userNamesSelected" key={index}>
                <span style={{ color: "#3472F7" }}>
                  {user.username}
                  <br />
                </span>
              </div>
            );
          })}
        </Modal.Body>
        <Modal.Footer className="justify-content-end">
          <Button
            variant="danger"
            className="mx-1 btn-fill"
            onClick={handleSaveofDelete}
          >
            Delete
          </Button>
          <Button
            variant="secondary"
            className="mx-1 btn-fill"
            onClick={handleCloseDeleteUser}
          >
            Close
          </Button>
        </Modal.Footer>
      </Modal>
    </>
  );
}

export default DeleteSelected;
