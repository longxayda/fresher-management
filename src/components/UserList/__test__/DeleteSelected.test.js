import React from "react";
import renderer from "react-test-renderer";
import DeleteSelected from "../DeleteSelected";
import Enzyme from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { shallow, mount } from "enzyme";

// Mock redux
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

Enzyme.configure({ adapter: new Adapter() });

describe("Test components AddUser", () => {
  const initialState = {
    UserList: {
      role: ["Admin"],
      selectedUser: [
        {
          username:'admin'
        }
      ],
    },
  };
  const mockStore = configureStore();
  let store = mockStore(initialState);

  test("Test render correctly without props", () => {
    const component = renderer.create(
      <Provider store={store}>
        <DeleteSelected />
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Test render correctly with VALID props", () => {
    const handleCurrentPage = jest.fn();
    const tree = mount(
      <Provider store={store}>
        <DeleteSelected
          showModal={true}
          handleCurrentPage={handleCurrentPage}
        />
      </Provider>
    );
    expect(tree.find("#deleteSelectedBtn").exists()).toEqual(true);
  });

  test("Test Show Modal", () => {
    const handleCurrentPage = jest.fn();
    const tree = mount(
      <Provider store={store}>
        <DeleteSelected
          showModal={true}
          handleCurrentPage={handleCurrentPage}
        />
      </Provider>
    );
    expect(tree.find("#titleDeletedSelected").exists()).toEqual(true);
  });

  test("Render span for check username", () => {
    const handleCurrentPage = jest.fn();
    const tree = mount(
      <Provider store={store}>
        <DeleteSelected
          showModal={true}
          handleCurrentPage={handleCurrentPage}
        />
      </Provider>
    );
    expect(tree.find("#userNamesSelected").exists()).toEqual(true);
  });
});
