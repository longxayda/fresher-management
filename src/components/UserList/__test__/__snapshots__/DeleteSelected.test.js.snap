// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test components AddUser Test render correctly without props 1`] = `
Array [
  <button
    className="btn-fill btn btn-danger"
    disabled={false}
    id="deleteSelectedBtn"
    onClick={[Function]}
    style={
      Object {
        "marginLeft": 10,
        "width": 130,
      }
    }
    type="button"
  >
    <i
      className="fas fa-trash"
      style={
        Object {
          "marginRight": "5px",
        }
      }
    />
    Delete
     
    1
  </button>,
  <div />,
]
`;
